scripts/quests/otherAreas/Unforgiven.lua

```diff
@@ -0,0 +1,104 @@
+-----------------------------------
+-- Unforgiven
+-----------------------------------
+-- Log ID: 4, Quest ID: 72
+-- Elysia :  !pos -50.410 -22.204 -41.640 26
+-- Pradiulot !pos -20.814 -22 8.399 26
+-- ???       !pos 110.714 -40.856 -53.154 26
+-----------------------------------
+require('scripts/globals/keyitems')
+require('scripts/globals/npc_util')
+require('scripts/globals/quests')
+require('scripts/globals/zone')
+require('scripts/globals/interaction/quest')
+-----------------------------------
+local ID = require("scripts/zones/Tavnazian_Safehold/IDs")
+-----------------------------------
+local quest = Quest:new(xi.quest.log_id.OTHER_AREAS, xi.quest.id.otherAreas.UNFORGIVEN)
+
+quest.reward =
+{
+    keyItem  = xi.ki.MAP_OF_TAVNAZIA,
+    gil      = 2000,
+    exp      = 2000,
+}
+
+quest.sections =
+{
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE
+        end,
+
+        [xi.zone.TAVNAZIAN_SAFEHOLD] =
+        {
+            ['Elysia'] = quest:progressEvent(200),
+
+            onEventFinish =
+            {
+                [200] = function(player, csid, option, npc)
+                    quest:begin(player)
+                end,
+            },
+        }
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.TAVNAZIAN_SAFEHOLD] =
+        {
+            ['Elysia'] =
+            {
+                onTrigger = function(player, npc)
+                    if not player:hasKeyItem(xi.ki.ALABASTER_HAIRPIN) then
+                        return quest:progressEvent(201)
```

Comment:
let's make this
```suggestion
                        return quest:event(201)
```
As it's not progressing the quest in any way. Same goes for anywhere else where the event doesn't increment any variables or progress the quest.

Reply:
what is the difference between progressEvent and just event? Is it only using progressEvent when I am stringing a couple different CS's together? Since the onEventFinish handles the progressing of the vars?

Reply:
progressEvent will override any other dialogue with this npc. so if it's involved in another quest, players won't be able to talk to progress other events with the NPC. Where event will allow for multiple different dialogues at any given moment with the NPC.

It basically future proofs our code. Some quests do require that quests are completed before anything else can be done with them. But it's not always the case.